controllers/novaexternalcompute_controller.go

```
@@ -184,6 +204,47 @@ func (r *NovaExternalComputeReconciler) Reconcile(ctx context.Context, req ctrl.
 	}
 	instance.Status.Hash[common.InputHashName] = inputHash
 
+	// we support stopping before deploying the compute node
+	// so only create the AAE CRs if we have deployment enabled.
+	if instance.Spec.Deploy {
+		// create all AEE resouce in parallel
+		libvirtAEE, err := r.ensureAEEDeployLibvirt(ctx, h, instance)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		novaAEE, err := r.ensureAEEDeployNova(ctx, h, instance)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// then check if they have compelted and requeue if not.
+		if libvirtAEE.Status.JobStatus != "Succeeded" {
+			return ctrl.Result{RequeueAfter: r.RequeueTimeout}, fmt.Errorf(
+				"libvirt deployment %s for %s", libvirtAEE.Status.JobStatus, instance.Name,
+			)
+		}
+		if novaAEE.Status.JobStatus != "Succeeded" {
+			return ctrl.Result{RequeueAfter: r.RequeueTimeout}, fmt.Errorf(
+				"nova deployment %s for %s", novaAEE.Status.JobStatus, instance.Name,
+			)
```

* We don't need to return an error here as this is not an error (and it produces nosy logging)
* We probably don't need to requeue here either as added `Owns(&aee.OpenStackAnsibleEE{}).` to the controller so we get reconciled when the status of the aee.OpenStackAnsibleEE changes.

---

yes your right, and ya i noticed the noices i was consideing changing it before but i wasnt sure what to replace this with.

i think i night just return normally.

im wondering if i should should log an info message or not.